classes/services/queryBuilders/SubmissionListQueryBuilder.inc.php

```diff
@@ -44,6 +44,38 @@ public function filterBySections($sectionIds) {
 	 * @return object Query object
 	 */
 	public function appGet($q) {
+		$primaryLocale = \AppLocale::getPrimaryLocale();
+		$locale = \AppLocale::getLocale();
+
+		$this->columns[] = Capsule::raw('COALESCE(stl.setting_value, stpl.setting_value) AS section_title');
```

Comment:
@asmecher Ok I tried to come up with a reproducible case of the issue. I have to say I don't really understand what's going on here too well. But let's take the following code [here](https://github.com/pkp/ojs/pull/1483/files#diff-65ff78d01cea089f9723f6d26ffdb1c9R56):

```php
		$q->leftJoin('section_settings as stpl', function($join) use($primaryLocale) {
			$join->on('s.section_id', '=', Capsule::raw("'stpl.section_id'"));
			$join->on('stpl.setting_name', '=', Capsule::raw("'section_title'"));
			$join->on('stpl.locale', '=', Capsule::raw("'{$primaryLocale}'"));
		});
```

My first attempt looked like this:

```php
		$q->leftJoin('section_settings as stpl', function($join) use($primaryLocale) {
			$join->on('s.section_id', '=', 'stpl.section_id');
			$join->where('stpl.setting_name', '=', 'section_title');
			$join->where('stpl.locale', '=', $primaryLocale);
		});
```

If you look at the SQL query and the bound params, they look like this:

```
select `s`.*, COALESCE(stl.setting_value, stpl.setting_value) AS section_title, COALESCE(sal.setting_value, sapl.setting_value) AS section_abbrev from `submissions` as `s` left join `section_settings` as `stpl` on `s`.`section_id` = `stpl`.`section_id` and `stpl`.`setting_name` = ? and `stpl`.`locale` = ? left join `section_settings` as `stl` on `s`.`section_id` = 'stl.section_id' and `stl`.`setting_name` = 'section_title' and `stl`.`locale` = 'en_US' left join `section_settings` as `sapl` on `s`.`section_id` = 'sapl.section_id' and `sapl`.`setting_name` = 'section_abbrev' and `sapl`.`locale` = 'en_US' left join `section_settings` as `sal` on `s`.`section_id` = 'sal.section_id' and `sal`.`setting_name` = 'section_abbrev' and `sal`.`locale` = 'en_US' where `s`.`context_id` = ? and `s`.`stage_id` in (?, ?, ?) and `s`.`section_id` in (?) group by `s`.`submission_id`, COALESCE(stl.setting_value, stpl.setting_value), COALESCE(sal.setting_value, sapl.setting_value) order by `s`.`date_submitted` desc

Array
(
    [0] => 1
    [1] => 1
    [2] => 2
    [3] => 3
    [4] => section_title
    [5] => en_US
    [6] => 2
)
```

So the `$join->where` statements were getting set as bound parameters, but were out of order with the eventual compilation of the query string.

The fix was to use multiple `$join->on` statements, like this:

```php
		$q->leftJoin('section_settings as stpl', function($join) use($primaryLocale) {
			$join->on('s.section_id', '=', 'stpl.section_id');
			$join->on('stpl.setting_name', '=', 'section_title');
			$join->on('stpl.locale', '=', $primaryLocale);
		});
```

For some reason, those `on` statements don't get set as bound parameters. They get rendered into the SQL statement. It's not clear to me whether the parameter binding/sanitizing is happening within the `leftJoin`, so it's still sanitizing the parameter. Of it it's just ignoring it and not doing any binding/sanitizing on it. That leads the the following SQL and params output:

```php
select `s`.*, COALESCE(stl.setting_value, stpl.setting_value) AS section_title, COALESCE(sal.setting_value, sapl.setting_value) AS section_abbrev from `submissions` as `s` left join `section_settings` as `stpl` on `s`.`section_id` = `stpl`.`section_id` and `stpl`.`setting_name` = `section_title` and `stpl`.`locale` = `en_US` left join `section_settings` as `stl` on `s`.`section_id` = 'stl.section_id' and `stl`.`setting_name` = 'section_title' and `stl`.`locale` = 'en_US' left join `section_settings` as `sapl` on `s`.`section_id` = 'sapl.section_id' and `sapl`.`setting_name` = 'section_abbrev' and `sapl`.`locale` = 'en_US' left join `section_settings` as `sal` on `s`.`section_id` = 'sal.section_id' and `sal`.`setting_name` = 'section_abbrev' and `sal`.`locale` = 'en_US' where `s`.`context_id` = ? and `s`.`stage_id` in (?, ?, ?) and `s`.`section_id` in (?) group by `s`.`submission_id`, COALESCE(stl.setting_value, stpl.setting_value), COALESCE(sal.setting_value, sapl.setting_value) order by `s`.`date_submitted` desc

Array
(
    [0] => 1
    [1] => 1
    [2] => 2
    [3] => 3
    [4] => 2
)
```

However, that leads to an OJS database error:

```
<h1>DB Error: Unknown column 'section_title' in 'on clause'</h1>
```

Finally, Kassim recommended the use of `Capsule::raw`, which seems to fix that problem:

```php
		$q->leftJoin('section_settings as stpl', function($join) use($primaryLocale) {
			$join->on('s.section_id', '=', Capsule::raw("'stpl.section_id'"));
			$join->on('stpl.setting_name', '=', Capsule::raw("'section_title'"));
			$join->on('stpl.locale', '=', Capsule::raw("'{$primaryLocale}'"));
		});
```

So it's not clear to me if this is really a case where order is causing problems. Maybe `where` methods are to be avoided inside `leftJoin` closures. It's not the only place I've run into this issue, but it's not exactly clear to me when the params get bound of order or why.

Maybe @kaschioudi has a better idea of a more specific example or use-case to illustrate the issue?